src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -541,10 +546,14 @@ public function addressShouldBeFilledAsShippingAddress(AddressInterface $address
      */
     public function iShouldBeOnTheCheckoutCompleteStep(): void
     {
-        Assert::inArray(
-            $this->getCheckoutState(),
-            [OrderCheckoutStates::STATE_PAYMENT_SKIPPED, OrderCheckoutStates::STATE_PAYMENT_SELECTED]
-        );
+        if ($this->sharedStorage->has('checkout_type') && $this->sharedStorage->get('checkout_type') === 'async') {
+            Assert::same($this->getCheckoutState(), AsynchronousOrderCheckoutStates::STATE_PAYMENT_SELECTED);
+        } else {
+            Assert::inArray(
+                $this->getCheckoutState(),
+                [OrderCheckoutStates::STATE_PAYMENT_SKIPPED, OrderCheckoutStates::STATE_PAYMENT_SELECTED]
+            );
+        }
```

Comment:
According to [this comment](https://github.com/Sylius/Sylius/pull/13793/files#r835206224) we should only use the async checkout in the API, so this if statement is not needed 🖖